sycl/source/detail/pi.cpp

```diff
@@ -131,6 +150,55 @@ vector_class<plugin> initialize() {
   // TODO: Correct the logic to store the appropriate plugin into GlobalPlugin
   // variable. Currently it saves the last plugin found.
   GlobalPlugin = std::make_shared<plugin>(PluginInformation);
+
+#ifdef XPTI_ENABLE_INSTRUMENTATION
+  // Not sure this is the best place to initialize
+  // the framework; SYCL runtime team needs to
+  // advise on the right place, until then we
+  // piggy-back on the initialization of the PI layer.
+  //
+  if (xptiTraceEnabled()) {
+    // Registers a new stream for 'sycl' and any
+    // plugin that wants to listen to this stream
+    // will register itself using this string or
+    // stream ID for this string.
+    //
+    auto StreamID = xptiRegisterStream(SYCL_STREAM_NAME);
```

Comment:
Why can't xpti framework register stream on the first call to some API? Why we need dedicated registering?

Reply:
StreamID is used by later calls to  xptiNotifySubscribers()